### PR TITLE
drivers: sensor: fxas21002: Update driver to use DEVICE_DT_GET

### PR DESCRIPTION
**Pull request by benjaminbjornsson**

### CODE REVIEW

**Review by gmarull — APPROVED** [review]

**Review by gmarull — DISMISSED** [review]

File: drivers/sensor/fxas21002/fxas21002_trigger.c [hunk 1]
```diff
@@ -16,13 +16,13 @@ static void fxas21002_gpio_callback(const struct device *dev,
 {
 	struct fxas21002_data *data =
 		CONTAINER_OF(cb, struct fxas21002_data, gpio_cb);
+	const struct fxas21002_config *config = dev->config;
```
Comment (MaureenHelm) [inline drivers/sensor/fxas21002/fxas21002_trigger.c:19]: `dev` is a GPIO device here, not a sensor device

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/fxas21002/fxas21002_trigger.c (hunk 1 above)
Comment (benjaminbjornsson) [reply drivers/sensor/fxas21002/fxas21002_trigger.c:19]: Nice catch! Should there be another `CONTAINER_OF` to get the sensor `dev` or is there a better solution?

Comment (benjaminbjornsson) [reply drivers/sensor/fxas21002/fxas21002_trigger.c:19]: Just realised that's not even possible... Not sure why, but I'm reluctant to keep a reference to `int_gpio` in `struct fxas21002_data`.

Comment (benjaminbjornsson) [reply drivers/sensor/fxas21002/fxas21002_trigger.c]: Done!

**Review by MaureenHelm — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**